qa/workunits/rbd/compare_mirror_image_alternate_primary.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+set -ex
+
+IMAGE=image
+RBD_MIRROR_MODE=snapshot
+MOUNT=/mnt/test
+WORKLOAD_TIMEOUT=5m
+
+. $(dirname $0)/rbd_mirror_helpers.sh
+
+launch_manual_msnaps() {
+  local cluster=$1
+  local pool=$2
+  local image=$3
+
+  for i in {1..30}; do
+    mirror_image_snapshot $cluster $pool $image
+    sleep 3s;
+  done
+}
+
+run_bench() {
+  local mountpt=$1
+  local timeout=$2
+
+  KERNEL_TAR_URL="https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.14.280.tar.gz"
+  sudo wget $KERNEL_TAR_URL -O $mountpt/kernel.tar.gz
+  sudo timeout $timeout bash -c "tar xvfz $mountpt/kernel.tar.gz -C $mountpt \
+    | pv -L 1k --timer &> /dev/null" || true
+}
+
+wait_for_non_primary_demoted_mirror_snap() {
+  local cluster=$1
+  local pool=$2
+  local image=$3
+
+  for s in 1 2 4 8 8 8 8 8 8 8 8 16 16; do
+    RET=$(rbd --cluster $cluster snap ls --all $pool/$image --format=json \
+            | jq 'last' \
+            | jq 'select(.name | contains("non_primary"))' \
+            | jq 'select(.namespace.state == "demoted")' \
+            | jq 'select(.namespace.complete == true)')
```

Comment:
Tested the following change to `wait_for_non_primary_demoted_mirror_snap()`
```
-  local ret
+  local demoted=false

   for s in 1 2 4 8 8 8 8 8 8 8 8 16 16; do
-    ret=$(rbd --cluster $cluster snap ls --all $pool/$image --format=json \
-            | jq 'last' \
-            | jq 'select(.name | contains("non_primary"))' \
-            | jq 'select(.namespace.state == "demoted")' \
-            | jq 'select(.namespace.complete == true)')
-    if [ "$ret" != "" ]; then
+    demoted=$(rbd --cluster $cluster snap ls --all $pool/$image --format=json \
+              | jq '(last | .name | startswith(".mirror.non_primary")) and
+                    (last | .namespace.state == "demoted") and
+                    (last | .namespace.complete == true)')
+
+    if [ "$demoted" = true ]; then
```

On testing, discovered that waiting for non primary demoted snapshot to be synced completely isn't sufficient to check that the image is ready to be promoted.  The test runs failed while trying to promote the image with being unable to create the promotion snapshot.
e.g., http://qa-proxy.ceph.com/teuthology/rraja-2024-01-30_17:01:34-rbd:mirror-main-distro-default-smithi/7539211/teuthology.log
In /a/rraja-2024-01-30_17:01:34-rbd:mirror-main-distro-default-smithi/7539211/remote/smithi175/log/ cluster2-client.mirror.29589.log.gz ,

```
2024-01-30T17:29:55.200+0000 7fd5ea73f640 10 librbd::ImageWatcher: 0x7fd5c8008a00 remote snap_create request: [4281,140556227323456,1] [mirror state=primary, complete=0, mirror_peer_uuids=05b2f9ef-e346-45e0-8d98-4a086cfaea16, clean_since_snap_id=head] .mirror.primary.07a17e35-3bf6-4b6e-bf88-133a68eabc73.61a582ff-e1d8-48f2-b699-b9c250763ba8 2
2024-01-30T17:29:55.200+0000 7fd5ea73f640 20 librbd::ExclusiveLock: 0x7fd5cc004190 accept_request=0 (request_type=0)
2024-01-30T17:29:55.200+0000 7fd5ea73f640 10 librbd::Watcher::C_NotifyAck 0x558063f59e90 finish: r=0
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::watcher::Notifier: 0x7fd5c8008a80 handle_notify: r=0
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::watcher::Notifier: 0x7fd5c8008a80 handle_notify: pending=0
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::image_watcher::NotifyLockOwner: 0x7fd5d0001e00 handle_notify: r=0
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::image_watcher::NotifyLockOwner: 0x7fd5d0001e00 handle_notify client responded with r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::ImageWatcher: remove_async_request: [4281,140556227323456,1]
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::Operations: handle_remote_request: r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::ImageState: 0x558063f568a0 handle_update_notification: refresh_seq = 1, last_refresh = 0
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::ImageState: 0x7fd5e00043b0 ImageUpdateWatchers::notify
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 15 librbd::mirror::snapshot::CreatePrimaryRequest: 0x7fd5c8002790 handle_create_snapshot: r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 -1 librbd::mirror::snapshot::CreatePrimaryRequest: 0x7fd5c8002790 handle_create_snapshot: failed to create mirror snapshot: (30) Read-only file system
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 15 librbd::mirror::snapshot::CreatePrimaryRequest: 0x7fd5c8002790 finish: r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 15 librbd::mirror::snapshot::PromoteRequest: 0x7fd5c801bc60 handle_create_promote_snapshot: r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 -1 librbd::mirror::snapshot::PromoteRequest: 0x7fd5c801bc60 handle_create_promote_snapshot: failed to create promote snapshot: (30) Read-only file system
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 15 librbd::mirror::snapshot::PromoteRequest: 0x7fd5c801bc60 finish: r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::mirror::PromoteRequest: 0x7fd5cc003550 handle_promote: r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 -1 librbd::mirror::PromoteRequest: 0x7fd5cc003550 handle_promote: failed to promote image: (30) Read-only file system
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::mirror::PromoteRequest: 0x7fd5cc003550 finish: r=-30
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::ImageState: 0x558063f568a0 handle_update_notification: refresh_seq = 2, last_refresh = 0
2024-01-30T17:29:55.200+0000 7fd5e9f3e640 20 librbd::ImageState: 0x7fd5e00043b0 ImageUpdateWatchers::notify
2024-01-30T17:29:55.200+0000 7fd5ed422e00 -1 librbd::api::Mirror: image_promote: failed to promote image
```

Instead of using `wait_for_non_primary_demoted_mirror_snap()` ,  Ilya suggested that we wait for the rbd mirror image status to be `up+unknown`,
```
wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${img} 'up+unknown'
wait_for_status_in_pool_dir ${CLUSTER2} ${POOL} ${img} 'up+unknown'
```
to be ready for promoting the image.